js/stores/windowStore.js

```diff
@@ -750,6 +752,22 @@ const doAction = (action) => {
         }
         break
       }
+    case windowConstants.WINDOW_ON_TOGGLE_BOOKMARK:
+      {
+        // show the AddEditBookmarkHanger control; saving/deleting takes place there
+        const activeFrame = frameStateUtil.getActiveFrame(windowState)
+        let siteDetail = siteUtil.getDetailFromFrame(activeFrame, siteTags.BOOKMARK)
+        const key = siteUtil.getSiteKey(siteDetail)
+
+        if (key !== null) {
```

Comment:
this logic here would be better if it was contained inside `siteUtil.getDetailFromFrame` method, IMO (especially since it can be tested).  There are other places that call this method and they too should be having the parentFolderId and customTitle set (in fact, if the other calls are not setting this, it may be causing a bug / bugs)